# Author: julian.morillo@bsc.es
# There is a known issue with these tests (see https://github.com/llvm/llvm-project/issues/35661)
# It looks that the behaviour is very architecture specific and it was somehow solved for aarch64
# but no mention to riscv64
diff -Naur openmp-ORIG/runtime/test/ompt/misc/control_tool.c openmp/runtime/test/ompt/misc/control_tool.c
--- openmp-ORIG/runtime/test/ompt/misc/control_tool.c	2026-07-21 17:40:02.223877889 +0200
+++ openmp/runtime/test/ompt/misc/control_tool.c	2026-07-21 17:42:20.095853061 +0200
@@ -2,6 +2,7 @@
 // RUN: %libomp-compile-and-run | FileCheck %s
 // REQUIRES: ompt
 // UNSUPPORTED: gcc-4, gcc-5, gcc-6, gcc-7
+// UNSUPPORTED: riscv64
 // clang-format on
 #define TEST_NEED_PRINT_FRAME_FROM_OUTLINED_FN
 #include "callback.h"
diff -Naur openmp-ORIG/runtime/test/ompt/synchronization/barrier/explicit.c openmp/runtime/test/ompt/synchronization/barrier/explicit.c
--- openmp-ORIG/runtime/test/ompt/synchronization/barrier/explicit.c	2026-07-21 17:40:02.231877853 +0200
+++ openmp/runtime/test/ompt/synchronization/barrier/explicit.c	2026-07-21 17:43:27.039836684 +0200
@@ -2,6 +2,7 @@
 // RUN: %libomp-compile-and-run | %sort-threads | FileCheck %s
 // REQUIRES: ompt
 // UNSUPPORTED: gcc-4, gcc-5, gcc-6, gcc-7
+// UNSUPPORTED: riscv64
 // clang-format on
 #include "callback.h"
 #include <omp.h>
diff -Naur openmp-ORIG/runtime/test/ompt/synchronization/critical.c openmp/runtime/test/ompt/synchronization/critical.c
--- openmp-ORIG/runtime/test/ompt/synchronization/critical.c	2026-07-21 17:40:02.231877853 +0200
+++ openmp/runtime/test/ompt/synchronization/critical.c	2026-07-21 17:44:12.703843576 +0200
@@ -2,6 +2,7 @@
 // RUN: %libomp-compile-and-run | FileCheck %s
 // REQUIRES: ompt
 // UNSUPPORTED: gcc-4, gcc-5, gcc-6, gcc-7
+// UNSUPPORTED: riscv64
 // clang-format on
 #include "callback.h"
 #include <omp.h>
diff -Naur openmp-ORIG/runtime/test/ompt/synchronization/flush.c openmp/runtime/test/ompt/synchronization/flush.c
--- openmp-ORIG/runtime/test/ompt/synchronization/flush.c	2026-07-21 17:40:02.231877853 +0200
+++ openmp/runtime/test/ompt/synchronization/flush.c	2026-07-21 17:44:48.351843228 +0200
@@ -3,6 +3,7 @@
 // REQUIRES: ompt
 // GCC generates code that does not call the runtime for the flush construct
 // XFAIL: gcc
+// UNSUPPORTED: riscv64
 // clang-format on
 
 #include "callback.h"
diff -Naur openmp-ORIG/runtime/test/ompt/synchronization/ordered.c openmp/runtime/test/ompt/synchronization/ordered.c
--- openmp-ORIG/runtime/test/ompt/synchronization/ordered.c	2026-07-21 17:40:02.231877853 +0200
+++ openmp/runtime/test/ompt/synchronization/ordered.c	2026-07-21 17:45:41.007840676 +0200
@@ -2,6 +2,7 @@
 // RUN: %libomp-compile-and-run | FileCheck %s
 // REQUIRES: ompt
 // UNSUPPORTED: gcc-4, gcc-5, gcc-6, gcc-7
+// UNSUPPORTED: riscv64
 // clang-format on
 #include "callback.h"
 #include <omp.h>
diff -Naur openmp-ORIG/runtime/test/ompt/synchronization/taskgroup.c openmp/runtime/test/ompt/synchronization/taskgroup.c
--- openmp-ORIG/runtime/test/ompt/synchronization/taskgroup.c	2026-07-21 17:40:02.231877853 +0200
+++ openmp/runtime/test/ompt/synchronization/taskgroup.c	2026-07-21 17:46:23.023836805 +0200
@@ -2,6 +2,7 @@
 // RUN:  %libomp-compile-and-run | %sort-threads | FileCheck %s
 // REQUIRES: ompt
 // UNSUPPORTED: gcc-4, gcc-5, gcc-6, gcc-7
+// UNSUPPORTED: riscv64
 // clang-format on
 
 #include "callback.h"
diff -Naur openmp-ORIG/runtime/test/ompt/synchronization/taskwait.c openmp/runtime/test/ompt/synchronization/taskwait.c
--- openmp-ORIG/runtime/test/ompt/synchronization/taskwait.c	2026-07-21 17:40:02.231877853 +0200
+++ openmp/runtime/test/ompt/synchronization/taskwait.c	2026-07-21 17:47:15.619828374 +0200
@@ -2,6 +2,7 @@
 // RUN: %libomp-compile-and-run | %sort-threads | FileCheck %s
 // REQUIRES: ompt
 // UNSUPPORTED: gcc-4, gcc-5, gcc-6, gcc-7
+// UNSUPPORTED: riscv64
 // clang-format on
 #include "callback.h"
 #include <omp.h>
